### PR TITLE
Readings dynamic update 

### PR DESCRIPTION
**Pull request by IuliaUngur**
cc: @asarbu

### CODE REVIEW

**Comment by IuliaUngur** [comment]
Issue #27

**Comment by IuliaUngur** [comment]
Issue #33

**Comment by IuliaUngur** [comment]
Issue #37

**Comment by IuliaUngur** [comment]
BUG: make predictions load in correct order

File: app/services/ap/creators_modifiers/prediction_creation.rb [hunk 1]
```diff
@@ -19,13 +21,14 @@ def prediction_sensors_with_result
       private
 
       def version_space
-        result = Ap::Algorithm::VersionSpace.new(@prediction).perform
-        @prediction.update_attribute(:result, result)
+        plant = @selection.present? ? @selection + " " : "plant "
```
Comment (iuliasincai) [inline app/services/ap/creators_modifiers/prediction_creation.rb]: You could extract this in a private method and pass its result to the `Ap::Algorithm::VersionSpace` service

File: app/assets/javascripts/react_components/live_prediction/react_live_prediction.js.jsx [hunk 2]
```diff
@@ -6,14 +6,21 @@ var ReactLivePrediction = React.createClass({
     predictions: React.PropTypes.array,
     access_id: React.PropTypes.number,
```
Comment (iuliasincai) [inline app/assets/javascripts/react_components/live_prediction/react_live_prediction.js.jsx]: Just as future idea, these react variables should be named using camel case.

File: app/controllers/predictions_controller.rb [hunk 3]
```diff
@@ -23,78 +23,69 @@ def live_hypotheses
...
     Prediction.predictions_without_sensors(environment).first.id
   end
 
-  def predictions_with_sensors(environment)
-    predictions = Prediction.predictions_with_sensors(environment).distinct
-
-    predictions.map { |prediction| prediction.sensor_result_set }
-  end
-
   def check_prerequisites
```
Comment (iuliasincai) [inline app/controllers/predictions_controller.rb]: How about using an `ActiveModel` object for these validations?

Comment (IuliaUngur) [inline app/controllers/predictions_controller.rb]: :+1: thanks for the idea. I think I'll implement this if I will have at least 1 more validation.

**Pull request merged**